### DISCUSSION TITLE
Zoom to elements

### DISCUSSION THREAD

**Discussion by ashish-srivastava1** [iTwin/itwinjs-core#2751] [General]
I am using the [zoomToElements ](https://www.itwinjs.org/reference/imodeljs-frontend/views/viewport/zoomtoelements/) API and setting the marginPercent in [ViewChangeOptions ](https://www.itwinjs.org/reference/imodeljs-frontend/views/viewchangeoptions/) parameter. The zoom operation should not get too close to the element in the viewer, that's why I am using the marginPercent. But it doesn't seem to work?

Are there other options to avoid zooming too close to an element in the viewer?

**Comment by pmconne** [comment]
Please define "doesn't work".

**Reply by bbastings** [reply]
You could try getting the element's range and either expand the range yourself, or supply the margin to "lookAtViewAlignedVolume".

note: marginPercent is no longer part of ViewChangeOptions in 3.0, it's moved to a "MarginOptions" that can be supplied to the lookAt methods.

**Comment by ashish-srivastava1** [comment]
> Please define "doesn't work".

I tried the following code (iTwinjs version: 2.19.0).

`const viewChangeOpts: ViewChangeOptions = {};
 viewChangeOpts.animateFrustumChange = true;
 viewChangeOpts.marginPercent = new MarginPercent(50, 50, 50, 50);
 const vp = IModelApp.viewManager.selectedView!;
 await vp.zoomToElements(zoomElementId, { ...viewChangeOpts });`

Whether I use or don't use marginPerecent property, the zoom result is same in both cases in my app. 

I used marginPercent to leave 50% view blank around the edges. So this is not happening.

**Reply by kabentley** [reply]
is the camera enabled? `MarginPercent` isn't applied when the camera is enabled.
